tools/bash/runfiles/runfiles.bash

```diff
@@ -122,16 +122,42 @@ function rlocation() {
         echo >&2 "INFO[runfiles.bash]: rlocation($1): looking in RUNFILES_MANIFEST_FILE ($RUNFILES_MANIFEST_FILE)"
       fi
       local -r result=$(grep -m1 "^$1 " "${RUNFILES_MANIFEST_FILE}" | cut -d ' ' -f 2-)
-      if [[ -e "${result:-/dev/null}" ]]; then
-        if [[ "${RUNFILES_LIB_DEBUG:-}" == 1 ]]; then
-          echo >&2 "INFO[runfiles.bash]: rlocation($1): found in manifest as ($result)"
-        fi
-        echo "$result"
-      else
+      if [[ -z "$result" ]]; then
+        # If path references a runfile that lies under a directory that itself
+        # is a runfile, then only the directory is listed in the manifest. Look
+        # up all prefixes of path in the manifest and append the relative path
+        # from the prefix if there is a match.
+        local prefix="$1"
+        local prefix_result=
+        local new_prefix=
+        while true; do
+          new_prefix="${prefix%/*}"
+          [[ "$new_prefix" == "$prefix" ]] && break
+          prefix="$new_prefix"
+          prefix_result=$(grep -m1 "^$prefix " "${RUNFILES_MANIFEST_FILE}" | cut -d ' ' -f 2-)
+          [[ -z "$prefix_result" ]] && continue
+          local -r candidate="${prefix_result}${1#"${prefix}"}"
+          if [[ -e "$candidate" ]]; then
+            if [[ "${RUNFILES_LIB_DEBUG:-}" == 1 ]]; then
+              echo >&2 "INFO[runfiles.bash]: rlocation($1): found in manifest as ($candidate) via prefix ($prefix)"
+            fi
+            echo "$candidate"
+            return 0
+          fi
+          # There is only ever one matching prefix in the runfiles manifest.
```

Comment:
Shouldn't this `break` statement be inside the if statement?

Reply:
I actually think it's correct as is, but agree this is confusing and thus added an extensive comment. Let me know what you think.

Reply:
Thanks! I missed the continue statement in the middle and thought the while loop would never run a second time. Thanks for the explanation!